prime-router/src/main/kotlin/fhirengine/translation/hl7/utils/ConstantResolver.kt

```diff
@@ -161,14 +158,8 @@ class FhirPathCustomResolver(private val customFhirFunctions: FhirPathFunctions?
                 emptyList()
             } else {
                 logger.trace("Evaluated FHIR Path constant $name to: $values")
-                // Convert string constants that are whole integers to Integer type to facilitate math operations
-                values.map {
-                    if (it is StringType && StringUtils.isNumeric(it.primitiveValue())) {
-                        IntegerType(it.primitiveValue())
-                    } else {
-                        it
-                    }
-                }
+                // previously we cast numbers to integers; see https://github.com/CDCgov/prime-reportstream/issues/12609
+                values
```

Comment:
We should use try/catch logic on the initial integer parsing code so that numbers that can be parsed as integers still are. This will avoid downstream effects if we do happen to be doing some sort of math operation.